### DISCUSSION TITLE
Playing while recording

### DISCUSSION THREAD

**Discussion by MishaMgla** [ant-media/Ant-Media-Server#6364] [Announcements]
I am trying to play a VOD while recording. However /v2/vods/{vodId} endpoint does not return filepath data till recording is stopped.

Is it possible to play record while it in recording status?

**Comment by Mohit-3196** [comment]
Hi @MishaMgla,
How are you?

It is an expected behavior when you record files with MP4. The MP4 recording is saved once the recording is stopped or the stream is ended after the MP4 muxing process is finished.

If you want to play the VoD when the stream is live, I recommend you use HLS to record the stream instead of MP4.
Here's how you can record with HLS https://antmedia.io/docs/guides/playing-live-stream/hls-playing/#save-hls-records

Let me know if you have any other queries.

--
Cheers,
Mohit

**Comment by burak-58** [comment]
Hi my friends,
Let me convert this to discussion. You can continue to discuss on this. If we need to implement or fix something, we can convert this to issue again.
